src/linear_algebra/free_module_pid.lean

```diff
@@ -404,6 +608,109 @@ noncomputable def module.free_of_finite_type_torsion_free' [module.finite R M]
   Σ (n : ℕ), basis (fin n) R M :=
 module.free_of_finite_type_torsion_free module.finite.exists_fin.some_spec.some_spec
 
+/-- If `M` is finite free over a PID `R`, then any submodule `N` is free
+and we can find a basis for `M` and `N` such that the inclusion map is a diagonal matrix.
```

Comment:
It's not just a diagonal matrix, it's in SNF. I think the docstring should also mention that.